docker/packager/gdb/Dockerfile

```diff
@@ -0,0 +1,30 @@
+# docker build -t clickhouse/gdb .
```

Comment:
Apparently this image should be created on docker hub first (`check_missing_images_on_dockerhub`), @Felixoid can you take a look at this PR and if it looks good help with this? (it is similar to https://github.com/ClickHouse/ClickHouse/pull/61011)

Reply:
@Felixoid kind ping

Reply:
I have been told that he is on a vacation. @maxknv can you help with this?

Reply:
> I have been told that he is on a vacation. @maxknv can you help with this?

let's build gdb in cctools image too? It looks like a good place for it and we don't need an extra image.

Reply:
`clickhouse/cctools` image is not included into any of our test images, its size is not an issue I believe
it serves us prebuild cctools when we build `clickhouse/binary-builder` and that's it:
`COPY --from=clickhouse/cctools:d9e3596e706b /cctools /cctools`
I think you want the same for gdb, right?

Reply:
Yes, but runner images do not need cctools, while they do need gdb

Reply:
so it's a separate issue (not connected to suggestion to use cctools for gdb build). right?
if so, what are runner images and why their sizes will bloat? didn't we just replace old gdb with a new gdb built from sources?

Reply:
I guess there is some misunderstanding here, let me try to explain

There are "runner" images:
- clickhouse/integration-test
- clickhouse/integration-tests-runner
- ...

They do not have toolchain (`clang`/...), but they do need `gdb`

You are suggesting to add gdb to `cctools`, which includes the compiler (it is based on `fasttest`), and not only it, but also some other stuff. But the problem with this that gdb has dynamic binary, so it depends from other libs, and this is tricky, **you need to know the list of files to copy from the `cctools` container.**

It maybe possible to build gdb statically I guess, but not that trivial - https://sourceware.org/pipermail/gdb-patches/2022-February/185721.html (see how much --disable flags it has), and this can be different gdb.

Reply:
OK, I think we could install gdb into a separate directory and copy the whole directory, though I don't like to put into `cctools` anyway since:
- it had been created specifically for apple cctools
- gdb and cctools should be updated separately

But if you insist, I will put it into this image

Reply:
Yes, there is a misunderstanding, and I'm afraid it is still not resolved.

I cannot understand how new `clickhouse/gdb` differs from `clickhouse/cctools` and why you think `clickhouse/gdb` works for building gdb and `clickhouse/cctools` does not. They both based on the same fasttest.

> you need to know the list of files to copy from the cctools container.

You need to know it anyway or how you are going to use gdb from `clickhouse/gdb`?

Reply:
>I cannot understand how new clickhouse/gdb differs from clickhouse/cctools and why you think clickhouse/gdb works for building gdb and clickhouse/cctools does not. They both based on the same fasttest.

The difference is in the name only, cctools is apple cctools specific

>You need to know it anyway or how you are going to use gdb from clickhouse/gdb?

This is not a concern anymore, will be installed into separate folder and the whole folder will be copied, small libraries will be installed from repos.

Reply:
> The difference is in the name only, cctools is apple cctools specific

If this is the only concern, i think cctools image still better than creating a new image. besidies its name it's perfectly fits our needs here.  I hope apple won't mind

Reply:
OK